lib/src/http/base_request.dart

```diff
@@ -20,8 +20,16 @@ import 'dart:convert';
 import 'package:fluri/fluri.dart';
 import 'package:http_parser/http_parser.dart';
 
+import 'package:w_transport/src/http/finalized_request.dart';
 import 'package:w_transport/src/http/request_dispatchers.dart';
+import 'package:w_transport/src/http/request_exception.dart';
 import 'package:w_transport/src/http/request_progress.dart';
+import 'package:w_transport/src/http/response.dart';
+
+typedef Future<Null> RequestInterceptor(BaseRequest request);
```

Comment:
I suppose this could work if I asserted that returned instance is the same instance as was passed in (I think `identical()` would work for that?). Would need to make sure the messaging/docs around that are clear.